src/transport_portals4.c

```diff
@@ -536,41 +535,29 @@ shmem_transport_startup(void)
     }
 #endif
 
-    desired = malloc(sizeof(ptl_process_t) * shmem_internal_num_pes);
-    if (NULL == desired) {
+    pe_map = malloc(sizeof(ptl_process_t) * shmem_internal_num_pes);
+    if (NULL == pe_map) {
         ret = 1;
         goto cleanup;
     }
 
-    ret = PtlGetPhysId(shmem_transport_portals4_ni_h, &my_id);
-    if (PTL_OK != ret) {
-        RETURN_ERROR_MSG("PtlGetPhysId failed: %d\n", ret);
-        goto cleanup;
-    }
-
     for (i = 0 ; i < shmem_internal_num_pes; ++i) {
+#ifdef USE_ON_NODE_COMMS
+        shmem_transport_addr_t addr = shmem_node_util_get_addr(i);
+        memcpy(&pe_map[i], &addr.ptl_addr, sizeof(ptl_process_t));
```

Comment:
I think you can get rid of this memcpy if you have `shmem_node_util_get_addr` take a pointer to the buffer where you want to store the addr.